crypto/bn256/cloudflare/curve.go

```diff
@@ -0,0 +1,229 @@
+package bn256
+
+import (
+	"math/big"
+)
+
+// curvePoint implements the elliptic curve y²=x³+3. Points are kept in Jacobian
+// form and t=z² when valid. G₁ is the set of points of this curve on GF(p).
+type curvePoint struct {
+	x, y, z, t gfP
+}
+
+var curveB = newGFp(3)
+
+// curveGen is the generator of G₁.
+var curveGen = &curvePoint{
+	x: *newGFp(1),
+	y: *newGFp(2),
+	z: *newGFp(1),
+	t: *newGFp(1),
+}
+
+func (c *curvePoint) String() string {
+	c.MakeAffine()
```

Comment:
This modifies `c`. I don't like it that printing a variable modifies it. I suggest : 
```golang
func (c *curvePoint) String() string {
	d := &curvePoint{}
	d.Set(c)
	d.MakeAffine()
	x, y := &gfP{}, &gfP{}
	montDecode(x, &d.x)
	montDecode(y, &d.y)
	return "(" + x.String() + ", " + y.String() + ")"
}

```

Reply:
This is what both the Google and Cloudflare code does. I'd rather not screw around with modifying the internals of a crypto library we don't understand.